mobile/lib/features/trade/trade_screen.dart

```diff
@@ -32,6 +31,66 @@ class _TradeScreenState extends State<TradeScreen> {
 
     const double tradeButtonWidth = 100.0;
 
+    SubmitOrderChangeNotifier submitOrderChangeNotifier =
+        context.watch<SubmitOrderChangeNotifier>();
+
+    if (submitOrderChangeNotifier.pendingOrder != null &&
+        submitOrderChangeNotifier.pendingOrder!.state == PendingOrderState.submitting) {
+      WidgetsBinding.instance.addPostFrameCallback((_) async {
+        return await showDialog(
+            context: context,
+            useRootNavigator: true,
+            builder: (BuildContext context) {
+              return Selector<SubmitOrderChangeNotifier, PendingOrderState>(
+                selector: (_, provider) => provider.pendingOrder!.state,
+                builder: (context, state, child) {
+                  const String title = "Submit Order";
+                  Widget body = Column(
+                    mainAxisSize: MainAxisSize.min,
+                    children: [
+                      SizedBox(
+                        width: 200,
+                        child: Wrap(
+                          runSpacing: 10,
+                          children: [
+                            ValueDataRow(
+                                type: ValueType.amount,
+                                value: submitOrderChangeNotifier.pendingOrderValues!.margin,
+                                label: "Margin"),
+                            ValueDataRow(
+                                type: ValueType.amount,
+                                value: submitOrderChangeNotifier.pendingOrderValues!.fee,
+                                label: "Fee")
+                          ],
+                        ),
+                      ),
+                      Padding(
+                        padding: const EdgeInsets.only(top: 20, left: 10, right: 10, bottom: 5),
+                        child: Text(
+                            "Your Position will be shown automatically in the Orders tab once your buy order has been filled!",
+                            style: DefaultTextStyle.of(context).style.apply(fontSizeFactor: 1.0)),
+                      )
+                    ],
+                  );
+
+                  switch (state) {
+                    case PendingOrderState.submitting:
+                      return SubmissionStatusDialog(
+                          title: title, type: SubmissionStatusDialogType.pending, content: body);
+                    case PendingOrderState.submittedSuccessfully:
+                      return SubmissionStatusDialog(
+                          title: title, type: SubmissionStatusDialogType.success, content: body);
+                    case PendingOrderState.submissionFailed:
+                      // TODO: This failure case has to be handled differently; are we planning to show orders that failed to submit in the order history?
```

Comment:
Even though, an order should not _fail_ in general but be pending, I think we should depict for now that a `market order` was not able to be filled immediately as discussed in our MVP scope.

Reply:
Thanks for pointing that out! I'll add a specific ticket for that task.